tests/cecil-tests/Documentation.cs

```diff
@@ -0,0 +1,229 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Xml;
+
+using NUnit.Framework;
+
+using Mono.Cecil;
+using Mono.Cecil.Cil;
+
+using ObjCRuntime;
+
+using Xamarin.Tests;
+using Xamarin.Utils;
+
+#nullable enable
+
+namespace Cecil.Tests {
+	[TestFixture]
+	public partial class Documentation {
+		// Verify that all our publicly visible APIs are documented.
+		// This is obviously not true, so we have a rather huge list of known failures.
+		// However, this will prevent us from adding more undocumented APIs by accident.
+		[Test]
+		public void VerifyEveryVisibleMemberIsDocumented ()
+		{
+			// We join all the APIs from all the platforms, so we can only run this test when all platforms are enabled.
+			Configuration.IgnoreIfAnyIgnoredPlatforms ();
+
+			// Collect everything
+			var xmlMembers = new HashSet<AssemblyApi> ();
+			var dllMembers = new HashSet<AssemblyApi> ();
+			foreach (var info in Helper.NetPlatformAssemblyDefinitions) {
+				// Just pick one of the implementation assemblies.
+				// We can't just list all the members in the ref assembly, because it doesn't contain any private members, and some private members are documented.
+				var implementationAssembly = Helper.NetPlatformImplementationAssemblyDefinitions.Where (v => Path.GetFileName (v.Path) == Path.GetFileName (info.Path)).First ();
+
+				var xml = GetDocumentedMembers (Path.ChangeExtension (info.Path, ".xml"));
+				xmlMembers.UnionWith (xml);
+				var dll = GetAssemblyMembers (implementationAssembly.Assembly);
+				dllMembers.UnionWith (dll);
+			}
+
+			// Propagate publicness
+			foreach (var dll in dllMembers) {
+				if (xmlMembers.TryGetValue (dll, out var xml)) {
+					xml.PubliclyVisible = dll.PubliclyVisible;
+				}
+			}
+
+			var documentedButNotPresent = xmlMembers.Except (dllMembers).ToList ();
+			Assert.Multiple (() => {
+				foreach (var doc in documentedButNotPresent)
+					Assert.Fail ($"{doc}: Documented API not found in the platform assembly. This probably indicates that the code to compute the doc name for a given member is incorrect.");
+			});
```

Comment:
Is it intended that we fail in the Multiple fails or do we want to add the following Assert.Fails in this multiple assert?